corehq/apps/app_manager/static/app_manager/js/widgets_v4.js

```diff
@@ -1,16 +1,21 @@
 // Note that this file exists only for select2 v4 and it depends on the paginate_releases URL being registered
 hqDefine("app_manager/js/widgets_v4", [
     'jquery',
+    'hqwebapp/js/assert_properties',
     'hqwebapp/js/initial_page_data',
     'select2/dist/js/select2.full.min',
 ], function (
     $,
+    assertProperties,
     initialPageData
 ) {
-    $(".app-manager-version-dropdown").each(function () {
-        $(this).select2({
+    var initVersionDropdown = function ($select, options) {
+        options = options || {};
+        assertProperties.assertRequired(options, [], ['url', 'width']);
```

Comment:
Just to confirm my understanding here, this means options should be present with url and width?

if thats true wont it make the check for options.url for ajax call redundant ?

Reply:
Oops, I used the wrong method. Updated to use `assert`, making it so url and width are both allowed but optional, and nothing other than url or width is allowed.